traitlets/traitlets.py

```diff
@@ -2689,3 +2602,101 @@ def info(self):
         if self.allow_none:
             return result + " or None"
         return result
+
+#-----------------------------------------------------------------------------
+# Casting TraitTypes
+#-----------------------------------------------------------------------------
+
+class CastingMixin(TraitType):
+    """A mixin class for TraitTypes which cast inputs
+
+    Attributes
+    ----------
+    cast_to : callable
+        The callable which will be casting inputs.
+    _cast_types : tuple of classes
+        An attribute that can be assigned in subclasses
+        to limit what types of values are allowed to be
+        cast (this attribute is not defined in the base
+        class for mixin purposes).
+    """
+
+    cast_to = None
+
+    def validate(self, obj, value):
+        cast_types = getattr(self, '_cast_types', All)
```

Comment:
This can be defined as the default on the mixin, rather than using getattr with a falllback